message SwitchInfoProto {
  optional int32 current_round = 1;
  optional int32 switching_round = 2;

  enum PartitionType {
    HYBRID = 1;
    FULL_GENUINE = 2;
    FULL_LOW_LATENCY = 3;
  };
  optional PartitionType partition_type = 3;

  enum MessageType {
    INIT_MSG = 1;
    ABORT = 2;
    GENUINE_SWITCH_ROUND_VOTE = 3;
    MEC_SYNCHRONIZED = 4;
    LOW_LATENCY_MAPPING_REQUEST = 5;
    LOW_LATENCY_MAPPING_RESPONSE = 6;
  };
  optional MessageType type = 4;

  // Key: partition id
  // Value: hop count from switching partitions.
  map<int32, int32> partition_mapping = 5;
}
